### ISSUE TITLE
Request for change in the CoapClient

### ISSUE THREAD

**Issue by Rajsmit** [eclipse/californium#92]
Hi,

We want to create a Request with setContentFormat()  and some data in the header using  "coapRequest.getOptions().addOption()" before calling observe. but the current exposed observe API only accept  Response handler and Accept option.

Could you please provide a way to pass the customized request in the observe API or make to below two API public.

`private CoapObserveRelation observeAndWait(Request request, CoapHandler handler)`

`private CoapObserveRelation observe(Request request, CoapHandler handler)`

Regards,
Rajesh Kumar.

**Comment by sophokles73** [comment]
You should be able to use the *advanced* methods of `CoapClient` for that:
````
Request req = Request.newGet().setURI("coap://server:5683/resourcename");
req.setObserve();
// now make all customizations, e.g. add options ...
req.getOptions().addOption(...);
...
CoapClient client = new CoapClient();
Response resp = client.advanced(req); // block while waiting for response
...
````

**Comment by Rajsmit** [comment]
Hi Kai,

Thanks for your reply.
The Observe API provides the CoapObserveRelation as return parameter that we want to use to cancel the observer based on some APP conditions. when we use the advance API then we will not get the CoapObserveRelation and this will require more large effort from our side.

Thanks,
Rajesh Kumar.

**Comment by sophokles73** [comment]
Oh, I see. That makes sense. I currently do not really see why we shouldn't make the more generic methods public. Would you like to create a PR for that?

**Comment by Rajsmit** [comment]
Hi Kai,

We created PR ( https://github.com/eclipse/californium/pull/94) for making observe API public.
Please check and let us know if it is fine.

Thanks,
Rajesh Kumar.

**Comment by sophokles73** [comment]
fixed in f6fa90d376749b651a6a6c8b4accb31bc6cd6eb6
